user_query.sql

```diff
@@ -0,0 +1,49 @@
+WITH observations_agg AS (
+    SELECT
+        t.user_id,
+        count(*) as obs_count,
+        sumMap(usage_details) as sum_usage_details,
+        sum(total_cost) as sum_total_cost
+    FROM
+        observations o
+            join traces t on t.id = o.trace_id
+            and t.project_id = o.project_id
```

Comment:
I read about using argMax for deduplication at query time this morning: https://clickhouse.com/blog/postgres-to-clickhouse-data-modeling-tips#use-argmax-to-deduplicate-rows-at-query-time. In that case, it may be applicable and avoid your need to use final. As there is a grouping anyway it should be fine as well.